### ISSUE TITLE
COREAUDIO_OpenDevice on iOS ignores sample rate

### ISSUE THREAD

**Issue by ivan-ushakov** [libsdl-org/SDL#9635]
Function [COREAUDIO_OpenDevice](https://github.com/libsdl-org/SDL/blob/SDL2/src/audio/coreaudio/SDL_coreaudio.m#L1050) has following code:

```cpp
AVAudioSession *session = [AVAudioSession sharedInstance];
[session setPreferredSampleRate:this->spec.freq error:nil];
this->spec.freq = (int)session.sampleRate;
```

It has several problems:

1. Does not check if it was successful so you don't know if sample rate was changed
2. Will fail to 48000 Hz in many cases

Is it possible to put this code under `#if` guard? Now if I use `SDL_OpenAudio` with sample rate 44100 Hz then I get 48000 Hz on my iPhone. If I remove that `setPreferredSampleRate` call then it works and I get 44100 Hz.

**Comment by icculus** [comment]
Do you _actually_ get 44100Hz if you don't explicitly set that? It's a weird choice of defaults, and it's even weirder that explicitly asking for it gets you something different than not asking for it, if it _is_ the default.

(We don't check for errors here because we don't care if it fails, since we have to take whatever frequency the session tells us in any case.)

**Comment by ivan-ushakov** [comment]
> Do you actually get 44100Hz

Yes, I check obtained spec from `SDL_OpenAudio()` call. Also I checked how that code works on my device, and it looks like `setPreferredSampleRate:` returns YES, but when I print `this->spec.freq` after this line:

```
this->spec.freq = (int)session.sampleRate;
printf("this->spec.freq=%d\n", this->spec.freq);
```

I see 48000 Hz.
